FEM-2105 IMADAI plugin support

giladna opened the pull request.

giladna committed:
DAI support

giladna committed:
Make DAI run again

giladna committed:
fix allAdsCompleted
fix ad progress in live case

giladna committed:
add preparation for isAlwaysStartWithPreroll

giladna committed:
check for empty string as well

giladna committed:
add protection for case no cuepoints available

giladna committed:
fix seek to end repeat

giladna committed:
support  adConfig.isAlwaysStartWithPreroll

giladna committed:
add missing implemntation

giladna committed:
make the start position apply to the stream media time

streamManager.getStreamTimeForContentTime(mediaConfig.getStartPosition());

giladna committed:
add missing events
fix autoplay/non autoplay
fix adCuePOints - add daiAdsList

giladna committed:
calculate getCuePoints once
fix warnings
refactor names

giladna committed:
fix type

giladna committed:
remove unneeded import

giladna committed:
FIX FEM-2382

after adding adController isPlaying API

giladna committed:
call get ad position once

giladna committed:
fix negative values for getPosition of ad in live mode

giladna committed:
add isLiveDAI on adConfig
protection add validation that for live ad we will use live in case given media is VOD

giladna committed:
support     public long getFakePlayerPosition(long realPlayerPosition) {

giladna committed:
fix seek is stuck

giladna committed:
fix duration position API's

fix cue points in change media

giladna committed:
roll back gradle comit

giladna committed:
extract cuepoints list  builder method

giladna committed:
fix FEM-2420 lock unlock while  ad loading DAI

giladna committed:
fix plugin name

giladna committed:
fix FEM-2418 the post-roll Ad doesn't started automatically after the media finsihed

giladna committed:
FIX FEM-2423 - fix playback stuck after lock device while ad loading in preroll and resume

giladna committed:
fix warnings

giladna committed:
tmp make ima complie

giladna committed:
rollback

giladna committed:
block autoplay on resume if not needed

giladna committed:
make constructor private and use factory instead

giladna committed:
clean commented code

giladna committed:
cosmetics

giladna committed:
move setValues to be inline

giladna committed:
code review changes

giladna committed:
use playerEngine load instead of prepare
fix if conditon to be more simple

noamtamim commented on imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java:
```diff
@@ -0,0 +1,1222 @@
+                boolean lastAdPlayed = true;
+                List<CuePoint> cuesList = streamManager.getCuePoints();
+                if (cuesList != null) {
+                    for (int i = 0 ; i < cuesList.size() ; i++) {
```
Use Iterator to iterate here.

giladna replied:
done

noamtamim commented on imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java:
```diff
@@ -0,0 +1,1222 @@
+                        if (!cuesList.get(i).isPlayed()) {
+                            allAdsPlayed = false;
+                        }
+                        if (i == cuesList.size() -1) {
```
Use iterator.hasNext() to detect the last item in the list.

giladna replied:
done

noamtamim commented on imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java:
```diff
@@ -0,0 +1,1222 @@
+        List<Long> cuePointsList = buildCuePointsList();
+        if (cuePointsList == null)
+            return;
+        if (cuePointsList.size() > 0) {
```
use List.isEmpty()

giladna replied:
done for all other same conditioons

noamtamim commented on imaplugin/src/main/java/com/kaltura/playkit/plugins/imadai/IMADAIPlugin.java:
```diff
@@ -0,0 +1,1222 @@
+        }
+    }
+
+//    private PKMediaConfig createMediaConfig(String url) {
```
Remove commented-out code.

giladna replied:
done

giladna committed:
code review fixes

working on stand alone SourceConfig

giladna committed:
code revire cont
we should not update the source config after coming back from bg since it will use the playerController source config

giladna committed:
expose api fro creating mediaSourceConfig not based on mediaConfig so media entry type will be set correctly
fix typo

giladna committed:
code review cont
No need in new mediaEntry

giladna committed:
remove adPluginType from AdRequested event
